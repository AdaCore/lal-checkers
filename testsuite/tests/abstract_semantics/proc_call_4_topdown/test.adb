procedure Test is
   type Point is record
      x : Integer;
      y : Integer;
      z : Integer;
   end record;

   function F(x : Integer) return Integer is
   begin
      if x < 42 then
         x := 10;
      else
         x := 20;
      end if;

      if x = 15 then
         return 123;
      end if;

      return x;
   end F;

   p : Point;
begin
   p.y := F(p.x);
end Ex1;
