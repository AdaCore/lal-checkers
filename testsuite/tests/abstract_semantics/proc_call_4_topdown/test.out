{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [-2147483648, 2147483647], [-2147483648, 2147483647])"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [10, 123], [-2147483648, 2147483647])"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [-2147483648, 2147483647], [-2147483648, 2147483647])"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [-2147483648, 2147483647], [-2147483648, 2147483647])"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [10, 20], [-2147483648, 2147483647])"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "p": "([-2147483648, 2147483647], [-2147483648, 2147483647], [-2147483648, 2147483647])"
        }
      }
    ]
  }
}
